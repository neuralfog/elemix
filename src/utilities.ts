import { activeRenderers } from './renderers';

export const ref = <Value>(
    value: Value | undefined = undefined,
): { value?: Value } => {
    return {
        value,
    };
};

export const fastUID = (): string =>
    Math.floor(performance.now() * 1000).toString(36) +
    Math.random().toString(36).slice(2, 6);

export function render(): Promise<boolean> {
    return new Promise((resolve) => {
        const check = () => {
            if (activeRenderers.size === 0) {
                resolve(false);
            } else {
                setTimeout(check, 0);
            }
        };
        check();
    });
}

export const camelToKebabCase = (input: string): string => {
    return (
        input.match(
            /[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|[0-9]*$)|[A-Z]?[a-z]+|[A-Z]|[0-9]+/g,
        ) || []
    )
        .map((x) => x.toLowerCase())
        .join('-');
};
