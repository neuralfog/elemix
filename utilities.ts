export { mergeClasses } from '@neuralfog/elemix-renderer/utils';
export * from './src/utilities';
