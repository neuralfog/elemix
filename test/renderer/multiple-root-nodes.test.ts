import { expect, test, describe, beforeEach } from 'vitest';
import { html } from '@neuralfog/elemix-renderer';
import { Present } from '@neuralfog/elemix-testing';
import { render } from '../../utilities';
import { HTML } from '@neuralfog/elemix-testing/snapshots';

import '../fixtures/renderer/MultilpleRootNodes.ts';

describe('Renderer Multiple Root Nodes', () => {
    beforeEach(() => {
        document.body.innerHTML = '';
    });

    test('render', async () => {
        const presenter = new Present().screen(html`<multiple-root-nodes />`);
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();
    });
});
