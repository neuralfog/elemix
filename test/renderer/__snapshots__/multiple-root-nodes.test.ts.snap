// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renderer Multiple Root Nodes > render 1`] = `
"
<multiple-root-nodes><!-- #shadowroot -->
  <div>Hello There!!</div>
  <div>Hello There!!</div>
  <div>Hello There!!</div>
</multiple-root-nodes>
"
`;
